LinkDotNet.ValidationExtensions/Dynamic/DynamicRangeAttribute.cs

```diff
@@ -0,0 +1,146 @@
+    public DynamicRangeAttribute(Type type, object minimum, string maximumPropertyName)
+        : base()
+    {
+        OperandType = type;
+        getMinimum = (ValidationContext validationContext) => minimum;
+        getMaximum = (ValidationContext validationContext) => GetActualValue(validationContext, OperandType, "Maximum", maximumPropertyName);
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DynamicRangeAttribute"/> class.
+    /// Allows for specifying range for arbitrary types. The minimum and maximum strings
+    /// will be converted to the target type.
+    /// </summary>
+    /// <param name="type">The type of the range parameters. Must implement IComparable.</param>
+    /// <param name="minimumPropertyName">The property-name of minimum.</param>
+    /// <param name="maximum">The maximum allowable value.</param>
+    public DynamicRangeAttribute(Type type, string minimumPropertyName, object maximum)
+       : base()
+    {
+        OperandType = type;
+        getMinimum = (ValidationContext validationContext) => GetActualValue(validationContext, OperandType, "Minimum", minimumPropertyName);
+        getMaximum = (ValidationContext validationContext) => maximum;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DynamicRangeAttribute"/> class.
+    /// Allows for specifying range for arbitrary types. The minimum and maximum strings
+    /// will be converted to the target type.
+    /// </summary>
+    /// <param name="type">The type of the range parameters. Must implement IComparable.</param>
+    /// <param name="minimumPropertyName">The property-name of minimum.</param>
+    /// <param name="maximumPropertyName">The property-name of maximum.</param>
+    public DynamicRangeAttribute(Type type, string minimumPropertyName, string maximumPropertyName)
+       : base()
+    {
+        OperandType = type;
+        getMinimum = (ValidationContext validationContext) => GetActualValue(validationContext, OperandType, "Minimum", minimumPropertyName);
+        getMaximum = (ValidationContext validationContext) => GetActualValue(validationContext, OperandType, "Maximum", maximumPropertyName);
+    }
+
+    /// <summary>
+    /// Gets the type of the <see cref="Minimum" /> and <see cref="Maximum" /> values
+    /// (e.g. Int32, Double, or some custom type).
+    /// </summary>
+    /// <value> The type of the <see cref="Minimum" /> and <see cref="Maximum" /> values. </value>
+    public Type OperandType { get; }
+
+    /// <inheritdoc />
+    protected override ValidationResult? IsValid(object? value, ValidationContext validationContext)
+    {
+        var minimumAsText = getMinimum.Invoke(validationContext).ToString();
+        var maximumAsText = getMaximum.Invoke(validationContext).ToString();
+
+        // Create RangeAttribute instance for validate value in actual range
+        var rangeAttribute = new RangeAttribute(OperandType, minimumAsText, maximumAsText);
```

Comment:
`minimumAsText` as well as `maximumAsText` are possibly null.

Either we check and throw if they are or if we are smarter than the compiler use the`!` operator on line 71/72

Reply:
Thanks, I added null-forgiving operator